flink-libraries/flink-ml/src/main/scala/org/apache/flink/ml/preprocessing/Splitter.scala

```diff
@@ -0,0 +1,210 @@
+   *                        of the dataset.  The DataSet is split into kFolds first, so that no
+   *                        observation will occurin in multiple folds.
+   * @param seed            Random number generator seed.
+   * @return An array of TrainTestDataSets
+   */
+  def kFoldSplit[T: TypeInformation : ClassTag](
+      input: DataSet[T],
+      kFolds: Int,
+      seed: Long = Utils.RNG.nextLong())
+    : Array[TrainTestDataSet[T]] = {
+
+    val fracs = Array.fill(kFolds)(1.0)
+    val dataSetArray = multiRandomSplit(input, fracs, seed)
+
+    dataSetArray.map( ds => TrainTestDataSet(ds,
+                                             dataSetArray.filter(_ != ds)
+                                                         .reduce(_ union _) ))
+
+  }
+
+  // --------------------------------------------------------------------------------------------
+  //  trainTestSplit
+  // --------------------------------------------------------------------------------------------
+  /**
+   * A wrapper for randomSplit that yields a TrainTestDataSet
+   *
+   * @param input           DataSet to be split
+   * @param fraction        Probability that each element is chosen, should be [0,1].
+   *                        This fraction refers to the training element in TrainTestSplit
+   * @param precise         Sampling by default is random and can result in slightly lop-sided
+   *                        sample sets. When precise is true, equal sample set size are forced,
+   *                        however this is somewhat less efficient.
+   * @param seed            Random number generator seed.
+   * @return A TrainTestDataSet
+   */
+  def trainTestSplit[T: TypeInformation : ClassTag](
+      input: DataSet[T],
+      fraction: Double = 0.6,
+      precise: Boolean = false,
+      seed: Long = Utils.RNG.nextLong())
+    : TrainTestDataSet[T] = {
+    val dataSetArray = randomSplit(input, fraction, precise, seed)
+    TrainTestDataSet(dataSetArray(0), dataSetArray(1))
+  }
+
+  // --------------------------------------------------------------------------------------------
+  //  trainTestHoldoutSplit
+  // --------------------------------------------------------------------------------------------
+  /**
+   * A wrapper for multiRandomSplit that yields a TrainTestHoldoutDataSet
+   *
+   * @param input           DataSet to be split
+   * @param fracArray       An array of three doubles, where the first element specifies the
+   *                        size of the training set, the second element the testing set, and
+   *                        the third element is the holdout set. These are proportional and
+   *                        will be normalized internally.
+   * @param seed            Random number generator seed.
+   * @return A TrainTestDataSet
+   */
+  def trainTestHoldoutSplit[T: TypeInformation : ClassTag](
+      input: DataSet[T],
+      fracArray: Array[Double] = Array(0.6,0.3,0.1),
```

Comment:
Why not requiring a `Tuple3[Double, Double, Double]` here, then we wouldn't have to do the check that the array has the correct length == one possibility where the user can shoot himself in the foot less ;-)